### PR TITLE
Catboost Model

### PR DESCRIPTION
**Pull request by miretteamin**
Issue #213

**Description**
I created 2 models of Catboost (developed by Yandex) with 2 different architectures, Model 1 was trained on 70% of the numerical dataset, it obtained 98.24% accuracy, and Model 2 was trained on 80% of the numerical dataset and it obtained 99.12% accuracy.

(This PR was created for part of issue #213 which is Catboost model implemented on the numerical dataset only)

 **Type of change**
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

 **Evaluation & Conclusion**
1- Checked on nan values and only column 33 (fractal_dimension_worst).
2- I tried to use MinMax scaler, Standard scaler and MinMax was better.
3- I tried to perform dimensionality reduction using PCA but the change wasn't significant.
4- Increasing the training dataset, gives a better accuracy when I increase the learning_rate and decrease the max_depth for each tree.
5- I tested the 2 models using the accuracy score implemented (built-in) in Catboost
*Model 1*
Model1

*Model 2*
Model2

6- Visualized the tests using a confusion matrix:
*Model 1*
Figure_1
*Model 2*
Figure_2

**Checklist**
- [x] My code follows the style guidelines of this project
- [x]  I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by freny24** [comment]
@miretteamin Hey , good work! But please convert your py file to ipynb file. It will be much easier to integrate into flask

**Comment by miretteamin** [comment]
> @miretteamin Hey , good work! But please convert your py file to ipynb file. It will be much easier to integrate into flask

Of course. Thanks!

**Comment by miretteamin** [comment]
> @miretteamin Hey , good work! But please convert your py file to ipynb file. It will be much easier to integrate into flask

I converted it.

**Comment by freny24** [comment]
@miretteamin Did you complete your image dataset part which was left?

**Comment by miretteamin** [comment]
> @miretteamin Did you complete your image dataset part which was left?

I'm still working on it.

**Comment by freny24** [comment]
@miretteamin okay do let me know , will merge after that

**Comment by srimanikantapalakollu** [comment]
@miretteamin Your PR has some merge conflicts please fix those merge conflicts as well. Good Work Keep it up. @freny24 Please help @miretteamin if any help is needed.

**Comment by miretteamin** [comment]
> @miretteamin Your PR has some merge conflicts please fix those merge conflicts as well. Good Work Keep it up. @freny24 Please help @miretteamin if any help is needed.

Yes I noticed, I'll fix it.
Thanks.

**Comment by miretteamin** [comment]
Hello @srimani-programmer  @freny24 , I finished the Catboost model on the images dataset that I took from the master branch as it doesn't exist in the New_Pipeline branch and I manipulated images names to be easier to sort and use in the model but actually, I don't know if I was allowed to do so or not, so, I'm going to push what I did so you can have a look on it and if I'm not allowed tell me and I'll remove the model and the dataset and I'll just push Catboost model on the numerical dataset.
Node on the new model: I couldn't get an accuracy of more than 90%, also, I asked a doctor about the images and he said that a lot of images needed more than one mask as there was more than one suspected tumor, maybe borders detection can be used to extract different suspected regions.
Thank you :)

**Comment by Suvraneel** [comment]
@srimani-programmer  @freny24 

Kindly get this PR sorted ASAP. :octocat:
